src/test/java/org/junitpioneer/jupiter/resource/SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static org.junitpioneer.jupiter.resource.Assertions.assertCanAddAndReadTextFile;
+import static org.junitpioneer.jupiter.resource.Assertions.assertReadableWriteableTemporaryDirectory;
+import static org.junitpioneer.jupiter.resource.Scope.SOURCE_FILE;
+
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class SecondSingleTopLevelTestMethodWithSourceFileScopedTempDirParameter {
```

Comment:
I think a comment explaining why this file is written the way it is would make it easier to grok them.

Reply:
Agreed, so similar to my [previous comment](https://github.com/junit-pioneer/junit-pioneer/pull/491#discussion_r953599706), I've moved this file into `TemporaryDirectoryTests.java` and commented all test cases to make it clearer why they're static.